src/java/org/apache/cassandra/index/sai/plan/StorageAttachedIndexSearcher.java

```diff
@@ -637,7 +667,36 @@ public PrimaryKeyIterator(UnfilteredRowIterator partition, Row staticRow, Unfilt
                       partition.isReverseOrder(),
                       partition.stats());
 
-                row = content;
+                assert !primaryKeysWithScore.isEmpty();
+                if (!content.isRow() || !(primaryKeysWithScore.get(0) instanceof PrimaryKeyWithScore))
+                {
+                    this.row = content;
+                    return;
+                }
+
+                // Clone the original Row
+                Row originalRow = (Row) content;
+                ArrayList<ColumnData> columnData = new ArrayList<>(originalRow.columnCount() + 1);
+                columnData.addAll(originalRow.columnData());
+
+                // inject +score as a new column
+                var tm = metadata();
+                var scoreColumn = ColumnMetadata.syntheticColumn(tm.keyspace,
+                                                                 tm.name,
+                                                                 ColumnMetadata.SYNTHETIC_SCORE_ID,
+                                                                 FloatType.instance);
+                var pkWithScore = (PrimaryKeyWithScore) primaryKeysWithScore.get(0);
+                columnData.add(BufferCell.live(scoreColumn,
+                                               FBUtilities.nowInSeconds(),
+                                               FloatType.instance.decompose(pkWithScore.indexScore)));
+
+                this.row = BTreeRow.create(originalRow.clustering(),
+                                           originalRow.primaryKeyLivenessInfo(),
+                                           originalRow.deletion(),
+                                           BTree.builder(ColumnData.comparator)
+                                                .auto(true)
+                                                .addAll(columnData)
+                                                .build());
```

Comment:
Do we need to skip this when `ANN_USE_SYNTHETIC_SCORE` is false? If we ignore it later on, I missed that.

Reply:
yes, there's code to ignore it later on

the problem is that wildcard column just returns true for any column when you ask "is this supposed to be fetched"

so add it and then ignore it later when the header complains seems like the way to go

that said i don't remember why I didn't try asking the header instead of the wildcard column, I will revisit this

Reply:
figured out how to ignore this correctly without the hassle of a version bump